File: project/.sdk/src/cmp/js/fragment/SdkError.fragment.js

const { Context } = require('./Context')


class ProjectNameError extends Error {

  isProjectNameError = true

  sdk = 'ProjectName'

  constructor(code, msg, ctx) {
    super(msg)
    this.code = code
    this.ctx = ctx
  }

}

module.exports = {
  ProjectNameError
}

File: project/.sdk/src/cmp/rb/fragment/SdkError.fragment.rb
# ProjectName SDK error

class ProjectNameError < StandardError
  attr_accessor :code, :msg, :sdk

  def initialize(code = "", msg = "")
    super(msg)
    @code = code
    @msg = msg
    @sdk = "ProjectName"
  end

  def error
    "#{@sdk}: #{@code}: #{@msg}"
  end
end
